File: core/src/com/jediterm/terminal/model/TerminalApplicationTitleListener.java
package com.jediterm.terminal.model;

import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;

public interface TerminalApplicationTitleListener {
  void onApplicationTitleChanged(@NotNull @Nls String newApplicationTitle);

  // Default empty implementation for backward compatibility
  default void onApplicationIconTitleChanged(@NotNull @Nls String newIconTitle) {
  }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/terminal/model/TerminalApplicationTitleListener.kt
package com.jediterm.terminal.model

import org.jetbrains.annotations.Nls

interface TerminalApplicationTitleListener {
    fun onApplicationTitleChanged(newApplicationTitle: @Nls String)

    fun onApplicationIconTitleChanged(newIconTitle: @Nls String) {
        // Default empty implementation for backward compatibility
    }
}
